script.js

```diff
@@ -32,8 +36,81 @@ const strokeWeight = 1.5;
 const pointRadius = 8;
 const fillOpacity = 0.7;
 
+let initialMapZoom = 4;
+//setting the initial zoom settings
+if (IS_MOBILE){
+  initialMapZoom = 3;}
+else if (IS_DESKTOP){
+  initialMapZoom = 5;}
+
+//initial values, if not given by the url
+let lat = 40.67;
+let long = -97.23;
+let z = initialMapZoom;
+let statesOn = true;
+let citiesOn = true;
+let countiesOn = true;
+
+//read url hash input
+let hash = location.hash;
+let input = inputValues(hash);
+
+// breaks up the url into an array; should be in the form
+// #/lat= & long= & z= & state= & cities= & counties=
+function inputValues(hash){
+  let inputVals = [];
+  let input = hash.slice(2).split("&");
+// splitting the hash by &, then creating an array
+  for (i=0;i<input.length;i++){
+    inputVals[input[i].split("=")[0]]=input[i].split("=")[1]
```

Comment:
@goldfarb you're treating `inputVals` as an Object, but using an Array. It would make more sense to do:

```js
  let inputVals = {};
  for (i=0; i<input.length; i++){
    let [key, value] = input.split;
    inputVals[key] = value;
  }
```
then you could just reference values like `inputVals.z` or `inputVals.lat`, etc.